### PR TITLE
Fixed bug in 'test_mesos.test_if_ucr_app_runs_in_new_pid_namespace'.

### PR DESCRIPTION
**Pull request by klueska**
Previously, there was a race between the `app_task` starting and it
writing its output to the `ps_output_file`. We now wait up to 10 seconds
for this file to appear before giving up and throwing an exception.

## High Level Description

What features does this change enable? What bugs does this change fix? High-Level description of how things changed.

## Related Issues

  - [DCOS_OSS-862](https://jira.mesosphere.com/browse/DCOS_OSS-862) Vagrant integration test test_mesos.test_if_ucr_app_runs_in_new_pid_namespace is flaky.

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: This is a fix to a test itself
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___

### CODE REVIEW

**Comment by adam-mesos** [comment]
@mesosphere-mergebot label [Ready For Review]

**Comment by karlkfi** [comment]
I'm having deja vu... how is this PR different than https://github.com/dcos/dcos/pull/1617 ?

**Comment by karlkfi** [comment]
Shouldn't we wait for the one on master then and then backport it? Why review it twice?

**Comment by adam-mesos** [comment]
Ideally, yes, we would land it in master first, then backport it.
But a) sometimes the patch release has more urgency, so we end up landing
it there first.
b) any cherry-pick conflicts when backporting would require a nominal
review to make sure the backporting didn't screw anything up.

File: packages/dcos-integration-test/extra/test_mesos.py [hunk 1]
```diff
@@ -157,7 +158,12 @@ def test_if_ucr_app_runs_in_new_pid_namespace(dcos_api_session):
         marathon_framework_id = dcos_api_session.marathon.get('/v2/info').json()['frameworkId']
         app_task = dcos_api_session.marathon.get('/v2/apps/{}/tasks'.format(app['id'])).json()['tasks'][0]
 
-        content = dcos_api_session.mesos_sandbox_file(
-            app_task['slaveId'], marathon_framework_id, app_task['id'], ps_output_file)
+        # There is a short delay between the `app_task` starting and it writing
+        # its output to the `pd_output_file`. Because of this, we wait up to 10
+        # seconds for this file to appear before throwing an exception.
+        @retrying.retry(wait_fixed=0.1, stop_max_delay=10000, retry_on_exception=lambda x: True)
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_mesos.py]: wait fixed is set to .1 milliseconds? The function being called here will do an HTTP GET, so I would suggest something like 1000ms or 2000ms.

Also, the default is to retry on all exceptions so you can just completely omit `retry_on_exception=lambda x: True`

Comment (klueska) [inline packages/dcos-integration-test/extra/test_mesos.py]: Yikes, that's a major oversight. Thanks for catching it! My intention was to wait for 100ms or 0.1s. The file should appear almost immediately, so I only expect to have to wait between 0 and 100ms for it to show up (i.e. only error out once), but I can set it at 1s if you think that's preferable.

**Pull request merged**